tools/online_editor/src/editor_widget.ts

```diff
@@ -87,7 +87,7 @@ class EditorPaneWidget extends Widget {
   #style = "fluent";
   #editor_documents: Map<string, ModelAndViewState>;
   #editor: monaco.editor.IStandaloneCodeEditor | null;
-  #keystroke_timeout_handle: number | undefined;
+  #keystroke_timeout_handle: number | NodeJS.Timeout | undefined;
```

Comment:
How would it be possible that we end up with a nodejs type here? This code runs only in the browser.

Reply:
Also, how do you reproduce the syntax error for this?

Reply:
Maybe a problem in how the linter is setup

Reply:
I finally figured that out:

Because cypress installed @types/node as one of its dependencies.

Tsc will unconditionally load all type information found in `@types` in the current node installation. So tsc saw two versions of `setTimeout` one for nodeJS and one for the browser (with different return types) and insisted that both need to be handled.

I forced tsc to ignore (most of) the types found in @types now. That also works around the bug where tsc finds contradicting definitions for AbortSignal when looking at web workers. So I was able to enable those again! Yeah!

Reply:
Excellent :)